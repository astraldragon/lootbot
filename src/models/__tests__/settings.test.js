const settings = require("../settings");

describe("models/settings", () => {
  beforeEach(() => {
    this.model = {
      belongsTo: jest.fn("belongsTo").mockReturnValue()
    };

    this.sequelize = {
      define: jest.fn("define").mockReturnValue(this.model)
    };

    this.Sequelize = {
      JSON: "json"
    };
  });

  it("should create the model definition", () => {
    settings(this.sequelize, this.Sequelize);
    expect(this.sequelize.define).toMatchSnapshot();
  });
});
