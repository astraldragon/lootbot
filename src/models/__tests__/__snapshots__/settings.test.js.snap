// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/settings should create the model definition 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Settings",
      Object {
        "settings": "json",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Object {
        "associate": [Function],
        "belongsTo": [MockFunction],
      },
    },
  ],
}
`;
